### ISSUE TITLE
[Survival][Suggestion]- Ore scanner Improvements

### ISSUE THREAD

**Issue by Crimson-Artist** [tsunamayo/Starship-EVO#5900]
Just a few QoL improvements to the ore scanner that I thought up while playing survival mode. I have made several mock ups to illustrate my point.

1. Have the name of the ore vein type appear during the scanner pulse. This will help player know what they are looking at as new players may be confused. To prevent screen clutter the name should only appear when the player's reticle hovers over the vein. 
![ore scanner suggestion 1](https://github.com/user-attachments/assets/ac651cd8-6766-4f1a-8d29-d9eed9745b23)
2. Allow any entity with a Antenna within range to appear during the pulse. Before I was able to build a vehicle I would some times get lost while out looking for resources. I had placed a antenna on my base for when I got a ship going but it is strange that we can only really make use of them when we are piloting a ship. My idea is that during the pulse the name a blip of any entity with a antenna will appear briefly then disappear. The range in which we see it will depend on the entity's "entity range" stat. 
![ore scanner suggestion 2](https://github.com/user-attachments/assets/7d352e73-d352-4df6-ae06-ce34d95daf45)
3. It also goes without saying but bares repeating that we need a ship scale version of the ore scanner. Perhaps have it set up like the antenna where just placing the block onto a ship allows us to press a non configurable hot key to activate the pulse. More ore scanner antennas will increase the range.

**Comment by tsunamayo** [comment]
@Crimson-Artist Hi,
So I though about 1) during my implementation, and I decided against it. It is just not natural with the way my mining patch generation works.
2) yes I wanted to do that also
3) yes I will work on a ship scanner (I said I would add it back in the day)
Thanks

**Comment by BigBadKangaroo** [comment]
> @Crimson-Artist Hi, So I though about 1) during my implementation, and I decided against it. It is just not natural with the way my mining patch generation works. 2) yes I wanted to do that also 3) yes I will work on a ship scanner (I said I would add it back in the day) Thanks

Hi there,

this is regarding to 1.) 
To be honest, I sometimes struggle to differentiate between different ores that have a simillar color like Iron and Silver. When the ore is quite far away I even have some issues with Coal, Iron and Silver (they have really simillar colors). Some sort of visual aid would be a nice QoL thing. Maybe we could be able to set custom colors for each ore in a menu? Or how vibrant we see the scanned ore patches?

**Comment by Dwarf-LordPangolin** [comment]
I agree with the issues some players are having with differentiating which resource is which. The scanner is a very useful and fun tool, but trial and error is required to learn which color = which element, and people will forget or mix colors up. This is especially true for players with color blindness.

I understand that the mining patch generation makes overlaying the kind of element difficult, but would it be possible to add the element symbol to the little circles for each resource? For example, looking at a patch of iron would show a bunch of little circles with "Fe" inside of them.

**Comment by ProPeach** [comment]
Elemental symbols sounds like a good option! It's a bit more sci-fi and more compact than the full name and would really help clear up the colour confusion

**Comment by tsunamayo** [comment]
Yeah I could try to add a texture of the element, but it would be impossible to read far away, and feels weird up close with a mega sign.
I could also simply improve the color a bit, add a legend if necessary.

**Comment by ProPeach** [comment]
I don't suppose the element symbol could be part of the HUD rather than an element in the world? That way the scaling could be kept more constant, I agree that having a huge sybol when you're up close would be very weird
